### PR TITLE
Adding client support for AttributeInTrunk variable

### PR DESCRIPTION
**Pull request by maxhutch**
# Citrine Python PR

## Description 
Adds client support for the AttributeInTrunk variable definition

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.

### CODE REVIEW

**Comment by maxhutch** [comment]
@ventura-rivera its before the process in the traversal order, but that is the reverse of chronological order, so I could see that being confusing.  If you provide a "mixing process" template, it will get the attributes from *later in the history* than that template.

I agree that the name is not great.  Do you have any other ideas?

**Comment by ventura-rivera** [comment]
@maxhutch Yes, I see what you are saying. That's really tricky. Another potential name that comes to mind is `AttributeInOutput` or `AttributeInProcessOutput`

**Comment by maxhutch** [comment]
I'll switch to AttributeInOutput for now and we can see how that goes.

File: src/citrine/ara/variables.py [hunk 1]
```diff
@@ -396,3 +398,70 @@ def __init__(self, *,
...
+
+class AttributeInOutput(
+        Serializable['AttributeInOutput'], Variable):
+    """[ALPHA] Attribute marked by an attribute template in the trunk of the history tree.
+
+    The search for an attribute that marks the given attribute template starts at the root
+    of the material history tree and proceeds until any of the given process templates are reached.
+    Those templates block the search from continuing into their ingredients but do not halt the
+    search entirely. This variable definition allows attributes that are present both in output
+    and the inputs of a process cto be distinguished.
```
Comment (bfolie) [inline src/citrine/ara/variables.py]:
```suggestion
    and the inputs of a process to be distinguished.
```

**Pull request merged**